### ISSUE TITLE
Slots

### ISSUE THREAD

**Issue by davidlesieur** [jpsca/jinjax#12]
I have just discovered JinjaX. I really like the super clean syntax, and the ease of passing attributes to components. Awesome work!

A feature analogous to the Web Components' [`<slot>` element](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_templates_and_slots) could be an interesting addition. 

Perhaps [django-components's implementation](https://github.com/EmilStenstrom/django-components/#using-slots-in-templates) can serve as inspiration. It supports conditional slots and nested slots.

**Comment by jpsca** [comment]
> I have just discovered JinjaX. I really like the super clean syntax, and the ease of passing attributes to components. Awesome work!

Thank you 😊

**Comment by rr-** [comment]
Another idea to go around this would be to have the ability to pass rendered components as parameters to child components.
For example:

```
<Main hero={<Hero>Buy our product, the <strong>{{ product_name }}</strong>!</Hero>}>
    We have the best product on the market yadda yadda
</Main>
````

**Comment by CiberNin** [comment]
It would be nice if I could create a Datagrid component with a fields slot that throws an error if it's passed children aren't all Field components.

**Comment by CiberNin** [comment]
> Another idea to go around this would be to have the ability to pass rendered components as parameters to child components. For example:
> 
> ```
> <Main hero={<Hero>Buy our product, the <strong>{{ product_name }}</strong>!</Hero>}>
>     We have the best product on the market yadda yadda
> </Main>
> ```

I'm not the biggest fan of parameter based slots.
Some libraries I've seen have been moving to component based slots.
```jinja
<Card>
    <Card.Header>
        My Card Title
    </Card.Header>
    <Card.Body>
        <div> Some Content </div>
    </Card.Body>
</Card>
```
We could achieve this in JinjaX by using call in preprocessing:
```jinja
{% call(slot) catalog.irender("Card", **{}) -%}
  {% if slot == 'Header' %}
    {% call(slot) catalog.irender("Card.Header", **{}) -%}
      My Card Title
    {%- endcall %}
  {% endif %}
  {% if slot == 'Body' %}
    {% call(slot) catalog.irender("Card.Body", **{}) -%}
      <div> Some Content </div>
    {%- endcall %}
  {% endif %}
{%- endcall %}
```

Although this would require a change in syntax when defining components. Can't just use "content" variable directly now.
`content(slot)`?
`content.slot`?
Has to internally call caller(slot). Can't pre-render content like we do now unless you have some way of scanning which slots are implemented in component. Which could be achievable if some syntax was added for defining them inside the `def` statement maybe.

**Comment by jpsca** [comment]
I ended following @CiberNin suggestion. Available since  [0.45](https://github.com/jpsca/jinjax/releases/tag/0.45)
